tools/authors/addbio.php

```diff
@@ -36,20 +38,20 @@
         if (isset($_POST['bio_id'])) {
             // edit existing bio
             $bio_id = get_integer_param($_POST, 'bio_id', null, null, null, TRUE);
-            $result = mysqli_query(DPDatabase::get_connection(), sprintf("
+            $result = DPDatabase::query(sprintf("
                 UPDATE biographies
                 SET bio = '%s'
-                WHERE bio_id = %s
-            ", mysqli_real_escape_string(DPDatabase::get_connection(), $bio), $bio_id));
+                WHERE bio_id = %d
+            ", DPDatabase::escape($bio), DPDatabase::escape($bio_id)));
```

Comment:
`$bio_id` is an int.

Reply:
You're escaping ints in various places, please review all of your changes and address these.